File: SDKs/amazonaws.com/gamelift/2015-10-01/go/pkg/models/operations/updatefleetcapacity.go
package operations

import (
	"openapi/pkg/models/shared"
)

type UpdateFleetCapacityXAmzTargetEnum string

const (
	UpdateFleetCapacityXAmzTargetEnumGameLiftUpdateFleetCapacity UpdateFleetCapacityXAmzTargetEnum = "GameLift.UpdateFleetCapacity"
)

type UpdateFleetCapacityHeaders struct {
	XAmzAlgorithm     *string                           `header:"style=simple,explode=false,name=X-Amz-Algorithm"`
	XAmzContentSha256 *string                           `header:"style=simple,explode=false,name=X-Amz-Content-Sha256"`
	XAmzCredential    *string                           `header:"style=simple,explode=false,name=X-Amz-Credential"`
	XAmzDate          *string                           `header:"style=simple,explode=false,name=X-Amz-Date"`
	XAmzSecurityToken *string                           `header:"style=simple,explode=false,name=X-Amz-Security-Token"`
	XAmzSignature     *string                           `header:"style=simple,explode=false,name=X-Amz-Signature"`
	XAmzSignedHeaders *string                           `header:"style=simple,explode=false,name=X-Amz-SignedHeaders"`
	XAmzTarget        UpdateFleetCapacityXAmzTargetEnum `header:"style=simple,explode=false,name=X-Amz-Target"`
}

type UpdateFleetCapacityRequest struct {
	Headers UpdateFleetCapacityHeaders
	Request shared.UpdateFleetCapacityInput `request:"mediaType=application/json"`
}

type UpdateFleetCapacityResponse struct {
	ConflictException           *interface{}
	ContentType                 string
	InternalServiceException    *interface{}
	InvalidFleetStatusException *interface{}
	InvalidRequestException     *interface{}
	LimitExceededException      *interface{}
	NotFoundException           *interface{}
	StatusCode                  int64
	UnauthorizedException       *interface{}
	UpdateFleetCapacityOutput   *shared.UpdateFleetCapacityOutput
}

File: SDKs/amazonaws.com/gamelift/2015-10-01/typescript/src/sdk/models/operations/updatefleetcapacity.ts
import { Metadata, SpeakeasyBase } from "../../../internal/utils/utils";
import * as shared from "../shared";

export enum UpdateFleetCapacityXAmzTargetEnum {
    GameLiftUpdateFleetCapacity = "GameLift.UpdateFleetCapacity"
}


export class UpdateFleetCapacityHeaders extends SpeakeasyBase {
  @Metadata({ data: "header, style=simple;explode=false;name=X-Amz-Algorithm" })
  xAmzAlgorithm?: string;

  @Metadata({ data: "header, style=simple;explode=false;name=X-Amz-Content-Sha256" })
  xAmzContentSha256?: string;

  @Metadata({ data: "header, style=simple;explode=false;name=X-Amz-Credential" })
  xAmzCredential?: string;

  @Metadata({ data: "header, style=simple;explode=false;name=X-Amz-Date" })
  xAmzDate?: string;

  @Metadata({ data: "header, style=simple;explode=false;name=X-Amz-Security-Token" })
  xAmzSecurityToken?: string;

  @Metadata({ data: "header, style=simple;explode=false;name=X-Amz-Signature" })
  xAmzSignature?: string;

  @Metadata({ data: "header, style=simple;explode=false;name=X-Amz-SignedHeaders" })
  xAmzSignedHeaders?: string;

  @Metadata({ data: "header, style=simple;explode=false;name=X-Amz-Target" })
  xAmzTarget: UpdateFleetCapacityXAmzTargetEnum;
}


export class UpdateFleetCapacityRequest extends SpeakeasyBase {
  @Metadata()
  headers: UpdateFleetCapacityHeaders;

  @Metadata({ data: "request, media_type=application/json" })
  request: shared.UpdateFleetCapacityInput;
}


export class UpdateFleetCapacityResponse extends SpeakeasyBase {
  @Metadata()
  conflictException?: any;

  @Metadata()
  contentType: string;

  @Metadata()
  internalServiceException?: any;

  @Metadata()
  invalidFleetStatusException?: any;

  @Metadata()
  invalidRequestException?: any;

  @Metadata()
  limitExceededException?: any;

  @Metadata()
  notFoundException?: any;

  @Metadata()
  statusCode: number;

  @Metadata()
  unauthorizedException?: any;

  @Metadata()
  updateFleetCapacityOutput?: shared.UpdateFleetCapacityOutput;
}
